projects/storefrontlib/src/cms-components/product/product-variants/guards/product-variant.guard.ts

```diff
@@ -16,40 +17,44 @@ import {
 export class ProductVariantGuard implements CanActivate {
   constructor(
     private productService: ProductService,
-    private routingService: RoutingService
+    private routingService: RoutingService,
+    private cmsService: CmsService
   ) {}
 
   canActivate(): Observable<boolean | UrlTree> {
     return this.routingService.getRouterState().pipe(
       map(state => state.nextState.params.productCode),
       switchMap((productCode: string) => {
-        if (Boolean(productCode)) {
-          return this.productService.get(productCode, ProductScope.VARIANTS);
-        } else {
-          return of(undefined);
-        }
-      }),
-      map((product: Product) => {
-        if (Boolean(product) && !product.purchasable) {
-          const variant = this.findVariant(product.variantOptions);
-          // below call might looks redundant but in fact this data is going to be loaded anyways
-          // we're just calling it earlier and storing
-          this.productService
-            .get(variant.code, ProductScope.LIST)
-            .pipe(
-              filter(Boolean),
-              take(1)
-            )
-            .subscribe((_product: Product) => {
-              this.routingService.go({
-                cxRoute: 'product',
-                params: _product,
-              });
-            });
-          return false;
-        } else {
-          return true;
+        // if open pdp from smartedit
+        if (this.cmsService.isLaunchInSmartEdit() && !Boolean(productCode)) {
```

Comment:
Do we really need to check for `!productCode`? IMO its a fair assumption that in SmartEdit we want to open any variant PDP, so this guard bypassed in SmartEdit. If so, then maybe we could simply perform the  check `if (this.cmsService.isLaunchInSmartEdit())` before the line 25 and return `of(true)` immediately?

btw. question: what's the advantage of using `if(Boolean(x))` over `if(x)`?

Reply:
In smartedit, SPA redirect to "default preview product" for PDP. So, if we only check `if (this.cmsService.isLaunchInSmartEdit())`, redirection will not happen, and `variantSelector` slot is empty. I tested it on smartedit.